src/PointwiseFunctions/AnalyticSolutions/Xcts/Schwarzschild.cpp

```diff
@@ -226,17 +521,90 @@ void SchwarzschildVariables<DataType>::operator()(
 template <typename DataType>
 void SchwarzschildVariables<DataType>::operator()(
     const gsl::not_null<tnsr::I<DataType, 3>*> shift_excess,
-    const gsl::not_null<Cache*> /*cache*/,
-    Tags::ShiftExcess<DataType, 3, Frame::Inertial> /*meta*/) const {
-  std::fill(shift_excess->begin(), shift_excess->end(), 0.);
+    const gsl::not_null<Cache*> cache,
+    Xcts::Tags::ShiftExcess<DataType, 3, Frame::Inertial> /*meta*/) const {
+  switch (coordinate_system) {
+    case SchwarzschildCoordinates::Isotropic: {
+      std::fill(shift_excess->begin(), shift_excess->end(), 0.);
+      break;
+    }
+    case SchwarzschildCoordinates::PainleveGullstrand: {
+      const auto& r =
+          get(cache->get_var(*this, detail::Tags::Radius<DataType>{}));
+      const DataType isotropic_prefactor = sqrt(2. * mass) / pow(r, 1.5);
+      *shift_excess = x;
+      get<0>(*shift_excess) *= isotropic_prefactor;
+      get<1>(*shift_excess) *= isotropic_prefactor;
+      get<2>(*shift_excess) *= isotropic_prefactor;
+      break;
+    }
+    case SchwarzschildCoordinates::KerrSchildIsotropic: {
+      const auto& r =
+          get(cache->get_var(*this, detail::Tags::ArealRadius<DataType>{}));
+      const auto& lapse =
+          get(cache->get_var(*this, gr::Tags::Lapse<DataType>{}));
+      const DataType isotropic_prefactor = 2. * mass * lapse / square(r);
+      *shift_excess = x;
+      get<0>(*shift_excess) *= isotropic_prefactor;
+      get<1>(*shift_excess) *= isotropic_prefactor;
+      get<2>(*shift_excess) *= isotropic_prefactor;
+      break;
+    }
+      // LCOV_EXCL_START
+    default:
+      ERROR("Missing case for SchwarzschildCoordinates");
+      // LCOV_EXCL_END
+  }
 }
 
 template <typename DataType>
 void SchwarzschildVariables<DataType>::operator()(
     const gsl::not_null<tnsr::ii<DataType, 3>*> shift_strain,
-    const gsl::not_null<Cache*> /*cache*/,
-    Tags::ShiftStrain<DataType, 3, Frame::Inertial> /*meta*/) const {
-  std::fill(shift_strain->begin(), shift_strain->end(), 0.);
+    const gsl::not_null<Cache*> cache,
+    Xcts::Tags::ShiftStrain<DataType, 3, Frame::Inertial> /*meta*/) const {
```

Comment:
(Opt) I just wanted to mention that I wasn't able to verify this implementation because I don't see a tensor equation for this referenced in the documentation. If you think this is something that is not obvious and worth documenting, then I would add it somewhere (e.g. add the equation to the class doc or reference another part of spectre or a textbook eq).